native/packages/react-native-bpk-component-phone-input/src/BpkDiallingCodeList.ios.js

```diff
@@ -0,0 +1,147 @@
+/*
+ * Backpack - Skyscanner's Design System
+ *
+ * Copyright 2018 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/* @flow */
+
+import { View, StyleSheet, SectionList } from 'react-native';
+import React from 'react';
+import BpkText from 'react-native-bpk-component-text';
+import {
+  colorGray50,
+  colorGray100,
+  spacingSm,
+  spacingMd,
+} from 'bpk-tokens/tokens/base.react.native';
+import { BpkDiallingCodeListItem } from '..';
+import {
+  type ListCommonProps,
+  LIST_COMMON_PROP_TYPES,
+  LIST_COMMON_DEFAULT_PROPS,
+} from './common-types';
+
+const spacing = spacingMd * 1.5;
+
+const styles = StyleSheet.create({
+  sectionHeader: {
+    paddingVertical: spacingSm,
+    paddingHorizontal: spacing,
+    backgroundColor: colorGray50,
+  },
+  separator: {
+    flex: 1,
+    height: 1,
+    backgroundColor: colorGray100,
+    marginLeft: spacing,
+  },
+});
+
+/*
+Takes data in the format:
+[
+  {name: 'Foo'},
+  {name: 'Bar'},
+  {name: 'Baz'},
+]
+
+and returns:
+{
+  f: [{name: 'Foo'}],
+  b: [{name: 'Bar'}, {name: 'Baz'}],
+}
+*/
+const createAlphabetisedList = data => {
```

Comment:
This can be simplified as

```javascript
_.groupBy(data, x => x.name[0].toLowerCase());
```

If we are okay with using lodash

Reply:
Good tip, thanks :)